src/config.rs

```diff
@@ -678,6 +723,27 @@ impl Default for DbOptions {
     }
 }
 
+/// An internal configuration for the database that contains instantiations
+/// of configuration objects defined in DbOptions
+#[derive(Clone)]
+pub(crate) struct DbConfig {
```

Comment:
Not crazy about this layering. Options vs. config vs. state is quite confusing to me. I normally think of config and options as immutable, but the clock is mutable.

WDYT about passing the `Arc<MonotonicClock>` directly into `DbInner`? We're already following that pattern with notifiers, stats, and so on?

Reply:
👍 works for me - I was imagining that the `DbConfig` is an instantiation of `DbOptions`, but I'm happy to just instantiate that within `DbInner` if it simplifies things